nixos/services/redis.nix

```diff
@@ -123,17 +112,32 @@ in
       };
     };
 
-    systemd.services.redis.serviceConfig.Restart = "always";
-
-    flyingcircus.activationScripts.redis = lib.stringAfter [ "fc-local-config" ] ''
-      if [[ ! -e /etc/local/redis/password ]]; then
-        ( umask 007;
-          echo ${lib.escapeShellArg password} > /etc/local/redis/password
-          chown redis:service /etc/local/redis/password
-        )
-      fi
-      chmod 0660 /etc/local/redis/password
-    '';
+    systemd.services.redis.serviceConfig =
+      let
+        preStart = pkgs.writeShellScript "redis-prestart" (
+          ''
+            if [[ ! -e /etc/local/redis/password ]]; then (
+            umask 007;
+          ''
+          + lib.optionalString (cfg.password == null) ''
+            ${pkgs.apg}/bin/apg -a 1 -M lnc -n 1 -m 32 > /etc/local/redis/password
```

Comment:
If redis is used in the mode were the password is assigned (by the deployment) in the nixos configuration, then we should implement this in a convergent manner, i.e. keep updating it even if the file exists. If the pw is not given, then only generate once, though.